src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1933,44 +1935,109 @@ GenTree* Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
     GenTree* tmp2 = nullptr;
     GenTree* tmp3 = nullptr;
 
-    bool   isConstant     = GenTreeVecCon::IsHWIntrinsicCreateConstant(node, simd32Val);
+    bool isConstant = false;
+    if (simdSize != 64)
+    {
+        isConstant = GenTreeVecCon::IsHWIntrinsicCreateConstant(node, simd32Val);
+    }
+    else
+    {
+        isConstant = GenTreeVecCon::IsHWIntrinsicCreateConstant(node, simd64Val);
+    }
     bool   isCreateScalar = (intrinsicId == NI_Vector128_CreateScalar) || (intrinsicId == NI_Vector256_CreateScalar);
     size_t argCnt         = node->GetOperandCount();
 
     if (isConstant)
     {
-        assert((simdSize == 8) || (simdSize == 12) || (simdSize == 16) || (simdSize == 32));
-
-        for (GenTree* arg : node->Operands())
+        if (simdSize != 64)
```

Comment:
Not sure this split is necessary. It duplicates quite a bit of code and it seems like the special-casing for `Vector512` could be limited behind a `if (simdSize == 64)` check just where needed.

Reply:
Done